datacube/index/memory/_metadata_types.py

```diff
@@ -18,12 +18,12 @@
 
 
 class MetadataTypeResource(AbstractMetadataTypeResource):
-    def __init__(self):
-        self.by_id = {}
-        self.by_name = {}
+    def __init__(self) -> None:
+        self.by_id: dict = {}
+        self.by_name: dict = {}
         self.next_id = 1
         for doc in default_metadata_type_docs():
-            self.add(self.from_doc(doc))
+            self.add(doc)
```

Comment:
The type of `default_metadata_type_docs()` is wrong, it's declared as `list[MetadataType]` but is actually `list[dict]`.

The `postgis` and `postgres` drivers are also following this code pattern of converting them with `self.from_doc`

Reply:
I don't know enough about the dynamic metadata type fields. There does appear to be a difference in the MetadataType that is created by the plain model, and the ones created in different drivers.

So as much as I'd prefer `default_metadata_type_docs()` code to change to match the Type, it'll be safer to just change the type to match the code. :(